tests/test_scatterwidgets.py

```diff
@@ -1,58 +1,404 @@
+    x_data, y_data, color_data, x_label, y_label, color_label = prepare_cont_test_data
+    plot_widget._onClick(None, y_data, color_data, "None: None", y_label, color_label)
+    plot_widget.plot()
+
+    assert plot_widget.x_label == "Count"
+    assert plot_widget.scatter is not None
+
+    plot_widget._onClick(x_data, None, color_data, x_label, "None: None", color_label)
+    plot_widget.plot()
+
+    assert plot_widget.y_label == "Count"
+    assert plot_widget.scatter is not None
+
+
+def test_hover_highlight_cont(plot_widget, prepare_cont_test_data):
+    """Test hover highlight functionality."""
+    x_data, y_data, color_data, x_label, y_label, color_label = prepare_cont_test_data
+    plot_widget._onClick(x_data, y_data, color_data, x_label, y_label, color_label)
+    plot_widget.plot()
+
+    plot_widget.update_hover_highlight(x_data["vec"][0], y_data["vec"][0])
+    assert plot_widget.hovered_point.data[0][0] == x_data["vec"][0]
+    assert plot_widget.hovered_point.data[0][1] == y_data["vec"][0]
+
+    plot_widget.update_hover_highlight(-1, -1)
+    assert plot_widget.data_point_label.text() == "Value: N/A"
+    assert plot_widget.hovered_point.data.size == 0
+
+
+def test_clear_hover_highlight(plot_widget, prepare_disc_test_data):
+    """Test clearing of hover highlight."""
+    x_data, y_data, color_data, x_label, y_label, color_label = prepare_disc_test_data
+    plot_widget._onClick(x_data, y_data, color_data, x_label, y_label, color_label)
+    plot_widget.plot()
+
+    plot_widget.update_hover_highlight(x_data["vec"][0], y_data["vec"][0])
+    assert plot_widget.hovered_point.data[0][0] == x_data["vec"][0]
+    assert plot_widget.hovered_point.data[0][1] == y_data["vec"][0]
+
+    plot_widget.clear_hover_highlight()
+    assert plot_widget.hovered_point.data.size == 0
+
+
+def test_toggle_drawing_mode(plot_widget):
+    """Test toggling of drawing mode."""
+    plot_widget.toggle_drawing_mode()
+    assert plot_widget.drawing is True
+    assert plot_widget.rectangle is False
+
+    plot_widget.toggle_drawing_mode()
+    assert plot_widget.drawing is False
+
+
+def test_toggle_rectangle_mode(plot_widget):
+    """Test toggling of rectangle mode."""
+    plot_widget.toggle_rectangle_mode()
+    assert plot_widget.rectangle is True
+    assert plot_widget.drawing is False
+
+    plot_widget.toggle_rectangle_mode()
+    assert plot_widget.rectangle is False
+
+
+def plot_coords_to_viewport(plot_widget, x, y):
+    plot_pos = pg.Point(x, y)
+    view_pos = plot_widget.scatter_plot.vb.mapViewToScene(plot_pos)
+    return plot_widget.scatter_plot.scene().views()[0].mapFromScene(view_pos)
+
+
+def send_mouse_event(event_type, widget, pos, button=QtCore.Qt.LeftButton):
+    """Send a mouse event to a widget."""
+    # necessary because of https://bugreports.qt.io/browse/QTBUG-5232
+    event = QtGui.QMouseEvent(event_type, pos, button, button, QtCore.Qt.NoModifier)
+    QtWidgets.QApplication.sendEvent(widget, event)
```

Comment:
This workaround is for macOS (as the issue from the url in the comment explains), and it could explain why the segfault appear for macOS, as it could be that this mechanism is not being taken into account by the qtbot.

Reply:
CC @Czaki, what do you think?